src/main/java/com/datastax/cdm/data/CqlData.java

```diff
@@ -63,6 +63,16 @@ public static boolean isCollection(DataType dataType) {
         return false;
     }
 
+    public static boolean isFrozen(DataType dataType) {
+        if (isPrimitive(dataType)) return false;
+        if (dataType instanceof UserDefinedType) return ((UserDefinedType) dataType).isFrozen();
+        if (dataType instanceof ListType) return ((ListType) dataType).isFrozen();
+        if (dataType instanceof SetType) return ((SetType) dataType).isFrozen();
+        if (dataType instanceof MapType) return ((MapType) dataType).isFrozen();
+        if (dataType instanceof TupleType) return dataType.asCql(true, false).contains("frozen<");
```

Comment:
Should we handle for ignore case here or will the output of [`asCql` is always in lowercase](https://docs.datastax.com/en/drivers/java/latest/com/datastax/oss/driver/api/core/type/DataType.html#asCql-boolean-boolean-)?

Reply:
This change comes from CDM-53, so I have incorporated the feedback on that branch. The v4 API is not specific about it, so using `toLowerCase()` is defensive and appropriate :) https://docs.datastax.com/en/drivers/java/4.13/com/datastax/oss/driver/api/core/CqlIdentifier.html#asCql-boolean-